### ISSUE TITLE
Missing aft-account-provisioning-customizations Step Function

### ISSUE THREAD

**Issue by rodalli** [aws-ia/terraform-aws-control_tower_account_factory#277]
**Terraform Version & Prov**: 1.2.8

**AFT Version**: 1.6.7

**Terraform Version & Provider Versions**

`terraform version`

```
Terraform v1.2.8
on darwin_amd64
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v4.38.0
+ provider registry.terraform.io/hashicorp/local v2.2.3
+ provider registry.terraform.io/hashicorp/random v3.4.3
+ provider registry.terraform.io/hashicorp/time v0.9.1
```

`terraform providers`

```
Providers required by configuration:
.
├── provider[registry.terraform.io/hashicorp/aws]
└── module.aft
    ├── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0, < 5.0.0
    ├── provider[registry.terraform.io/hashicorp/local]
    ├── module.aft_account_provisioning_framework
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    ├── module.aft_backend
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    ├── module.aft_code_repositories
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   └── provider[registry.terraform.io/hashicorp/local]
    ├── module.aft_feature_options
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    ├── module.aft_iam_roles
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   ├── module.ct_management_service_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   ├── module.log_archive_exec_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   ├── module.log_archive_service_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   ├── module.aft_exec_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   ├── module.aft_service_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   ├── module.audit_exec_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   ├── module.audit_service_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   └── module.ct_management_exec_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    ├── module.packaging
    │   └── provider[registry.terraform.io/hashicorp/archive]
    ├── module.aft_account_request_framework
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   └── provider[registry.terraform.io/hashicorp/time]
    ├── module.aft_customizations
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   └── provider[registry.terraform.io/hashicorp/local]
    ├── module.aft_ssm_parameters
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
    │   └── provider[registry.terraform.io/hashicorp/random]
    └── module.aft_lambda_layer
        ├── provider[registry.terraform.io/hashicorp/aws] >= 4.9.0
        ├── provider[registry.terraform.io/hashicorp/random]
        └── provider[registry.terraform.io/hashicorp/local]

Providers required by state:

    provider[registry.terraform.io/hashicorp/local]

    provider[registry.terraform.io/hashicorp/aws]

    provider[registry.terraform.io/hashicorp/archive]

    provider[registry.terraform.io/hashicorp/random]

    provider[registry.terraform.io/hashicorp/time]
```

**Bug Description**

A strange problem... I've just deployed AFT to a brand new Organization with Control Tower. I _believe_ I've met all of the prereqs, and didn't run into any issues or errors when applying the AFT Terraform. After that I ran through the post-deployment steps in the AWS documentation here: https://docs.aws.amazon.com/controltower/latest/userguide/aft-post-deployment.html

Creating new accounts seems to work fine and they get joined to Control Tower and SSO access configured as expected. However, the account customizations are not working and I don't see CodePipeline pipelines getting created for these as I would expect. When I investigated the Step Functions that I believe are responsible for this (`aft-account-provisioning-framework`) I found all executions of the pipeline had failed with the following error:

```
{
  "resourceType": "states",
  "resource": "startExecution.sync:2",
  "error": "StepFunctions.StateMachineDoesNotExistException",
  "cause": "State Machine Does Not Exist: 'arn:aws:states:us-east-1:<aft-management-account-id>:stateMachine:aft-account-provisioning-customizations' (Service: AWSStepFunctions; Status Code: 400; Error Code: StateMachineDoesNotExist; Request ID: ab45d80e-80aa-45bd-afff-2ae077b2e65d; Proxy: null)"
}
```

Sure enough, the referenced Step Function isn't there. The only ones that are created in the AFT account are:

- `aft-invoke-customizations`
- `aft-feature-options`
- `aft-account-provisioning-framework`

I've gone back to look at the AFT Terraform, and it does not seem to create this Step Function. Does this get created by another pipeline? Any idea what the problem is here?

**To Reproduce**
1. Add an account in the `aft-account-request` repo with an `account_customizations_name` parameter like so:

`aft-account-request//terraform/test-account.tf`:
```
module "test_account_01" {
  source = "./modules/aft-account-request"

  control_tower_parameters = {
    AccountEmail = "email+unique_alias@gmail.com"
    AccountName  = "Test 01"
    ManagedOrganizationalUnit = "Test OU"
    SSOUserEmail     = "email@gmail.com"
    SSOUserFirstName = ""
    SSOUserLastName  = ""
  }

  change_management_parameters = {
    change_requested_by = "me"
    change_reason       = "Add account customizations parameter"
  }

  custom_fields = {}

  account_customizations_name = "test-account-01"
}
```
2. Create a matching directory and TF config for the `account_customizations_name` value in the `aft-account-customizations` repo:

`aft-account-customizations//test-account-01/terraform/main.tf`:
```
locals {
  tags = {
    GithubRepo = "aft-account-customizations"
    GithubOrg  = "me"
    ManagedBy  = "AFT"
  }
}

################################################################################
# VPC Module
################################################################################

module "vpc-ue2-01" {
  source = "terraform-aws-module/vpc/aws"
  # Declare the custom provider using the alias from ./aft-providers.jinja
  provider = aws.us_east_2

  name = "test-ue2"
  cidr = "10.0.0.0/18"

  azs             = ["us-east-2a", "us-east-2b"]
  private_subnets = ["10.0.0.0/22", "10.0.4.0/22"]
  public_subnets  = ["10.0.32.0/24", "10.0.33.0/24"]

  enable_ipv6 = true

  enable_nat_gateway     = true
  single_nat_gateway     = false
  one_nat_gateway_per_az = false

  public_subnet_tags = {
    Name = "test-ue2-public"
  }

  tags = local.tags

  vpc_tags = {
    Name = "test-ue2"
  }
}
```

3. Execute the Step Functions state machine `aft-invoke-customizations` in the AFT account with the following input:

```
{
  "include": [
    {
      "type": "all"
    }
  ]
}
```

**Expected behavior**
The Step Function would succeed and create/initiate a CodePipeline to apply the above Terraform to the `Test 01` account

**Related Logs**

When I investigated the Step Functions that I believe are responsible for this (`aft-account-provisioning-framework`) I found all executions of the pipeline had failed with the following error:

```
{
  "resourceType": "states",
  "resource": "startExecution.sync:2",
  "error": "StepFunctions.StateMachineDoesNotExistException",
  "cause": "State Machine Does Not Exist: 'arn:aws:states:us-east-1:<aft-management-account-id>:stateMachine:aft-account-provisioning-customizations' (Service: AWSStepFunctions; Status Code: 400; Error Code: StateMachineDoesNotExist; Request ID: ab45d80e-80aa-45bd-afff-2ae077b2e65d; Proxy: null)"
}
```

**Comment by rodalli** [comment]
I'm a complete dummy!

I figured out that the Step Function is created by templates/files in the `aft-account-provisioning-customization` repo, and then realized the issue was that I was missing some files in there that I hadn't completely populated from the source repos yet.

Once I got those files in place and pushed to the repo, I re-executed the `aft-invoke-customizations` state machine and everything worked as expected. :)
